### PR TITLE
Reduced lung vtu input

### PR DESCRIPTION
**Pull request by maxiludwig**
# Read the reduced lung tree from a VTU mesh file

## Description and Context

The reduced lung module did not use mesh based geometry input but reads its geometry from several input field json files. 

This PR replaces that with a proper mesh file. The geometry, the element types, the per-element
data and the boundary-condition assignment now all come from the VTU mesh, and everything in the
input file that had to scale with the tree is gone.

**Before / after**, for `reduced_lung_3_aw_2_tu.4C.yaml`:

```yaml
# before
lung_tree:
  topology:
    num_nodes: 6
    num_elements: 5
    node_coordinates: {from_file: "reduced_lung_3_aw_2_tu_fields.json"}
    element_nodes:    {from_file: "reduced_lung_3_aw_2_tu_fields.json"}
  element_type:       {from_file: "reduced_lung_3_aw_2_tu_fields.json"}
  generation:         {from_file: "reduced_lung_3_aw_2_tu_fields.json"}
  airways:
    radius:           {from_file: "reduced_lung_3_aw_2_tu_fields.json"}
boundary_conditions:
  num_conditions: 3
  bc_type:      {constant: Pressure}
  bc_node_id:   {from_file: "reduced_lung_3_aw_2_tu_fields.json"}
  boundary_value:
    bc_function_id: {from_file: "reduced_lung_3_aw_2_tu_fields.json"}
```

```yaml
# after
geometry:
  file: "vtu/reduced_lung_3_aw_2_tu.vtu"
lung_tree:
  airways:
    element_blocks: [1]
    radius: {from_mesh: "radius"}
  terminal_units:
    element_blocks: [2]
boundary_conditions:
  pressure:
    - id: 1
      function_id: 1
    - id: 2
      function_id: 2
```

Three mechanisms carry the change:

- **Element types come from cell blocks.** `airways` and `terminal_units` each declare the ids of
  the cell blocks they own, the way every other problem type in 4C identifies its elements. Every
  cell block must be claimed by exactly one element type.
- **Per-element data comes from cell data.** With element types resolved from the mesh, all
  remaining per-element input (`radius`, …) can be given as `from_mesh` and is read from the cell
  data of the lung mesh. `generation` is dropped from the input entirely — it stays in the mesh as
  cell data, but nothing in the model reads it.
- **Boundary conditions attach through `bc_id` point data.** A node with `bc_id == 0` is
  unconstrained; a node with `bc_id == N` carries the condition defined under `id: N`. Conditions
  are therefore identified by plain integers, just like cell blocks, and the input file holds one
  reusable definition per id instead of one entry per constrained node.

The mesh is read with `Core::IO::VTU::read_vtu_file()` directly rather than through the generic
multi-format dispatch of `MeshReader`, to keep the changes inside `reduced_lung`.

### Commit structure

Four commits, meant to be read in order:

1. **`Give list() its own explicit store`** — the only `src/core/io` change. `list()` always nested
   its parsed `InputParameterContainer::List` under its own name in the parent container, so
   getting a list into a struct member required wrapping it in an enclosing
   group. Adds a `store` field to `ListData` and reports the store target through `stores_to`. The
   new `boundary_conditions` section needs this: without it, `group<BoundaryConditions>` rejects
   its own lists as storing to `InputParameterContainer`. The commit also fixes a latent bug in
   `ListSpec::set_default_value()`, which cast the storage by value so the default list was
   silently discarded — see the commit message, and adds a regression test covering a defaulted, fixed-size list with an explicit store.

2. **`Read lung topology from external mesh file`** — replaces `lung_tree.topology` with
   `geometry.file` and builds the discretization from the mesh points and line2 cells. Element
   lengths are now derived from the discretization instead of from topology input fields, so the
   airway and terminal-unit registries take the reference length as an argument.

3. **`Take lung element types and BCs from the mesh`** — cell blocks, `from_mesh` fields and
   `bc_id`, as described above.

4. **`Migrate reduced lung tests to VTU meshes`** — replaces the `reduced_lung_*_fields.json`
   fixtures with VTU meshes under `tests/input_files/vtu/`.

Note that commits 2 and 3 change the input format while the test inputs are still on the old one;
they are migrated in commit 4. The branch is therefore intended to be reviewed and merged as a
whole and is not bisectable through the middle. The commit boundaries are there to make the
reasoning reviewable, not to give four independently working states.

### Notes for reviewers

Deliberate, user-visible changes:

- **Constant boundary values are gone.** A constant is just a constant function, so
  `boundary_value` now only takes a `function_id`. No test input used `bc_value`, so nothing needed
  migrating, but external input files will.
- **A node can no longer carry a pressure and a flow condition at once**, since it carries exactly
  one `bc_id`. No input file made use of that.
- **Reading a mesh requires VTK**, so the migrated tests declare it as a dependency.

One point worth a deliberate decision rather than drift: the reduced lung identifies cell blocks
with a flat `element_blocks: [1]`, whereas the standard 4C mechanism
(`ELEMENT_BLOCKS` under `<FIELD> GEOMETRY`) is a list of `{ID | NAME}` entries and supports
name-based identification. The flat list is simpler and matches how the ids appear in the mesh.

## Disclosure of AI assistance

Developed with Claude Code (Claude Opus 5), used for implementation, review and test migration; the
individual commits carry `Co-Authored-By` trailers. The design decisions were made by the author. All AI-assisted code
has been reviewed and validated by the author, who takes responsibility for the contribution.

### CODE REVIEW

File: src/reduced_lung/src/4C_reduced_lung_helpers.cpp [hunk 1]
```diff
@@ -220,79 +228,277 @@ namespace ReducedLung
...
     Core::FE::DiscretizationBuilder<3> builder(discretization.get_comm());
 
-    const int my_rank = Core::Communication::my_mpi_rank(discretization.get_comm());
-    if (my_rank == 0)
+    const int num_nodes = static_cast<int>(node_coordinates.size());
+    for (int node_id = 0; node_id < num_nodes; ++node_id)
     {
-      if (topology.num_nodes <= 0)
+      builder.add_node(node_coordinates[node_id], node_id, nullptr);
+    }
```
[the remark this answers is not in the document]
Comment (maxiludwig) [reply src/reduced_lung/src/4C_reduced_lung_helpers.cpp:242]: `DiscretizationBuilder::build()` `(src/core/fem/src/discretization/4C_fem_discretization_builder.cpp:239)` only converts the local nodes_/elements_ maps into the `RawMesh` when `my_rank == 0`; everything a non-root rank added is simply never read.

**Review by bgoderbauer — DISMISSED** [review]
Thank you for improving the input so much! The changes look surprisingly very minimalistic.

**Review by davidrudlstorfer — COMMENTED** [review]
@maxiludwig thanks a lot for tackling this after the discussion 😁 I think this looks really really great and the input files look and feel much cleaner :)

Just one question; back then when the unit tests were introduced there were some changes made so the .json files would work for the unit tests. Can we revert those changes again?

**Comment by rjoussen** [comment]
I don't know anything about reduced lung, so this is mainly out of curiosity: I am wondering whether it would be possible to align the reduced lung geometry input with the (external) geometry input for all the other fields, to have something more like this:
```yaml
REDUCED LUNG GEOMETRY:
  FILE: vtu/reduced_lung_3_aw_2_tu.vtu"
  SHOW_INFO: "detailed_summary"
  ELEMENT_BLOCKS:
    - NAME: airways
      REDUCED_LUNG:
        AIRWAYS:
          RADIUS: {from_mesh: "radius"}
          ...
    - NAME: terminal_units
      REDUCED_LUNG:
        TERMINAL_UNITS:
          RHEOLOGICAL MODEL: ...
```
I think the reduced lung specific element types could register themselves as normal element types for the reduced lung physics via `setup_element_definition`.

**Comment by maxiludwig** [comment]
@davidrudlstorfer as discussed offline, there were no tweaks that made their way to the `main` to enable unit testing for the reduced lung json input.

@rjoussen I would argue that reduced lung is non using real FEM elements (we have e.g. no shape functions). We require only a geometry, nothing more. So i would prefer to separate it from the FE elements.

**Review by bgoderbauer — APPROVED** [review]

**Review by davidrudlstorfer — APPROVED** [review]

**Comment by rjoussen** [comment]
> @rjoussen I would argue that reduced lung is non using real FEM elements (we have e.g. no shape functions). We require only a geometry, nothing more. So i would prefer to separate it from the FE elements.

Thanks for the explanation!
